### ISSUE TITLE
backend_client.go:333: Could not send request

### ISSUE THREAD

**Issue by grickard** [strukturag/nextcloud-spreed-signaling#80]
Hello,

System: LXC Ubuntu 18.04
Talk: 9.0.8

After adding my signaling server in the Talk settings and getting an OK status with signaling server version, I am unable to start calls. When I attempt to start the call the connection simply times out. I have this entry in my signaling servers log:

> backend_client.go:333: Could not send request {"type":"auth","auth":{"version":"1.0","params":{"userid":"grickard","ticket":"Yx75v4urLPeA9pXU:1612983382:grickard:4bedb6a87b9bd618f49025218909f7943e8aee2a0d1952730110b81b06703061"}}} to https://nc.example.com/ocs/v2.php/apps/spreed/api/v1/signaling/backend: Post https://nc.example.com/ocs/v2.php/apps/spreed/api/v1/signaling/backend: dial tcp <my_ip>:443: i/o timeout

I'm not sure if the fact that I built the standalone signaling server in LXC has anything to do with the issue but I cannot find any signs to indicate as such. Coturn, Janus and Nginx are all functioning fine with no errors being logged.

The NC application servers are NAT'd to their public IP, however the signaling server has it's public IP attached to the vNIC.

Thanks!

UPDATE:

I'm getting another error when attempting to use the Android Talk app:

> Error reading from <IP_ADDRESS>.239: read tcp 127.0.0.1:8080->127.0.0.1:54622: use of closed network connection

**Comment by fancycode** [comment]
> dial tcp <my_ip>:443: i/o timeout

This indicates a network problem on the VM of the signaling server when trying to connect to your Nextcloud server. Can you ping this ip manually or perform a simple `curl` request?

**Comment by grickard** [comment]
Thanks for the reply.

I am successfully able to ping the server and perform a curl request both inside and outside.

curl https://signaling.spt.cloud/standalone-signaling/api/v1/welcome I get the following response:

> HTTP/1.1 200 OK
> Server: nginx/1.14.0 (Ubuntu)
> Date: Thu, 11 Feb 2021 17:43:08 GMT
> Content-Type: application/json; charset=utf-8
> Content-Length: 94
> Connection: keep-alive
> Strict-Transport-Security: max-age=63072000; includeSubdomains; preload
> 
> {"nextcloud-spreed-signaling":"Welcome","version":"61000f5f698a9bdf6f5a5ea2c56f0d412fac55c3"}

Could it have anything to do with my application servers sitting behind a router with 1:1 NAT?

**Comment by grickard** [comment]
UPDATE:

I went ahead and spooled up a fresh Nextcloud instance in a fresh VM and put the VM outside the firewall and bridged a public IP directly to it's net interface. This seems to have eliminated the "Could not send request" error. 

Unfortunately still when I go to start the call it just spins and never starts. The only message in the log is:

> signaling[1187]: client.go:251: Client from <IP_ADDRESS> has RTT of 29 ms (29.063595ms)

Which I assume is a normal entry. There is nothing pertaining to Talk or Spreed in the app instances log. The only thing I can think of trying at this point is rebuilding the signaling server on a KVM machine.

**Comment by grickard** [comment]
Last night I rebuilt the signaling server in a KVM machine and successfully got everything working today.

So for anyone who stumbles across this in the future, the HPB signaling server will NOT work correctly in an unprivileged LXC container. I haven not tried with a privileged container as this is a security compromise.

I also had trouble with application servers that were sitting behind a router with 1:1 NAT. I was able to get everything working by putting my application servers NICs outside the firewall and configuring them to directly use a public IP.
